### PR TITLE
Major cascade upgrades

### PR DESCRIPTION
**Pull request by notlee**
Upgrade to the latest version of all Origami components.
Gets us up to date and brings some nice little visual improvements,  
including improved contrast and layout of form inputs, links, buttons.

A selection of screenshots (doesn't show all pages)...

before
Screenshot 2020-04-06 at 17 28 21

after:
Screenshot 2020-04-06 at 17 25 59

before:
Screenshot 2020-04-06 at 17 30 38

after:
Screenshot 2020-04-06 at 17 29 59

before (core experience):
Screenshot 2020-04-06 at 17 29 05

after (core experience):
Screenshot 2020-04-06 at 17 29 07

### CODE REVIEW

File: src/.eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,3 @@
-'use strict';
-
 module.exports = {
-	"env": {
-		"browser": true
-	}
-}
+	"extends": "origami-component"
```
Comment (JakeChampion) [inline src/.eslintrc.js:2]: If all JavaScript files in the. `src` folder are for the client-side/browser, we should move to using `export`/`import` and not `module.exports`/`require` :-)

File: src/scss/_status-labels.scss [hunk 2]
```diff
@@ -9,38 +9,44 @@
...
 @include oLabelsAddState('dead', (
-	background-color: oColorsGetPaletteColor('black')
+	background-color: oColorsByName('black')
 ));
 
 @include oLabelsAddState('master', (
-	background-color: oColorsGetPaletteColor('wheat')
+	// master brand "wheat" colour
+	// (not available to internal brands)
+	background-color: #f2dfce
```
Comment (JakeChampion) [inline src/scss/_status-labels.scss:26]: I guess we should look to use other colours that are within the internal brand colours?

File: views/partials/overview/sidebar.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="registry__container__sticky">
+<div class="o-layout-sticky-sidebar-container">
```
Comment (JakeChampion) [inline views/partials/overview/sidebar.html:1]: 👍

File: src/scss/overview/_search-form.scss [hunk 4]
```diff
@@ -1,36 +1,33 @@
...
+	));
+}
 
-	&--loading &-search-loading {
-		display: block;
-	}
+// registry loading class is defined twice intentionally.
+// o-loading mixins create a new selector
+// after the one they are used in, to output
+// rules after keyframes
```
Comment (JakeChampion) [inline src/scss/overview/_search-form.scss:14]: I don't understand what this is trying to explain 😔

File: lib/routes/components.js [hunk 5]
```diff
@@ -261,7 +261,7 @@ module.exports = app => {
 				versions,
 				currentBrand,
 				nav: 'index',
-				heading: 'Demos',
+				heading: component.type === 'imageset' ? 'Imageset' : 'Demos',
```
Comment (JakeChampion) [inline lib/routes/components.js:264]: 👍

File: src/scss/overview/_search-form.scss (hunk 4 above)
Comment (notlee) [reply src/scss/overview/_search-form.scss:14]: `oLoadingContent` outputs css at the root using `@at-root` and creates a new selector.
So custom styles that follow `oLoadingContent` do not come after the mixin output 😞 
https://github.com/Financial-Times/o-loading/blob/082a9c5e35f1fa61b3adce755b1df5193b72e0c8/src/scss/_mixins.scss#L79

Does that make more sense as a comment?

File: src/scss/_status-labels.scss (hunk 2 above)
Comment (notlee) [reply src/scss/_status-labels.scss:26]: I'm not sure what's best to do here and settled for this for now.

I considered adding it to o-colors for internal brands to reference but I have
before seen an abomination of a UI with a wheat/slate internal mix when
we were first working on branding -- thanks to design/project dev miscommunication.
So I'm keen to limit the internal brand palette and make it clear wheat shouldn't be
used generally.

This is maybe a unique usecase as we are using the label to signify the master brand,
so we kinda do want it wheat, although we shouldn't _really_ worry too much if it gets 
out of sync with actual wheat for this usecase.

The other hexes here which were previously generated using master brand palettes
are replicating a Sass-like pink and a JS-like yellow. We could set custom colour
usecases for these that use the actual Sass/JS colours. But I figured 
master-brand-esqe versions are nice and we don't care if they get out of sync
if the alternative is to use a non-palette colour anyhow :)

File: src/.eslintrc.js (hunk 1 above)
Comment (notlee) [reply src/.eslintrc.js:2]: 👍 I cleared up the remaining `module.exports`

File: src/scss/_status-labels.scss (hunk 2 above)
Comment (notlee) [reply src/scss/_status-labels.scss:34]: I added the above to the code comments, and I removed the use of sky which I didn't have a justification for :)

File: src/scss/_status-labels.scss [hunk 6]
```diff
@@ -36,17 +44,31 @@
...
 
 @include oLabelsAddState('js', (
+	// This is creating a JS-link yellow colour
+	// using a mix of mandarin and the master
+	// lemon colour. Lemon is not available to
+	// internal brands but we don't really care
+	// about them staying in sync for this usecase.
+	// In-fact, we could replace this with a custom
+	// colour outside the palette, of the actual
+	// JS logo yellow (personal preference)..
```
Comment (notlee) [inline src/scss/_status-labels.scss:54]:
```suggestion
	// This is creating a JS-link yellow colour
	// using a mix of mandarin and lemon.
	// We could replace this with a custom
	// colour outside the palette, of the actual
	// JS logo yellow (personal preference)...
```

File: src/scss/_status-labels.scss [hunk 7]
```diff
@@ -9,38 +9,66 @@
...
 	background-color: oColorsMix('mandarin', 'lemon', 30)
 ));
 
 @include oLabelsAddState('scss', (
-	background-color: oColorsMix('candy', 'velvet', 50)
+	// This is creating a Sass-link pink
```
Comment (notlee) [inline src/scss/_status-labels.scss:59]:
```suggestion
	// This is creating a Sass-like pink
```

File: src/scss/_status-labels.scss (hunk 2 above)
Comment (JakeChampion) [reply src/scss/_status-labels.scss:34]: Excellent explanation!

File: src/scss/overview/_search-form.scss (hunk 4 above)
Comment (JakeChampion) [reply src/scss/overview/_search-form.scss:14]: That does make more sense 👍

**Pull request merged**